### PR TITLE
Add login

### PR DESCRIPTION
**Pull request by herulume**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -205,13 +205,7 @@ captures/
...
 # IntelliJ
 *.iml
-.idea/workspace.xml
-.idea/tasks.xml
-.idea/gradle.xml
-.idea/assetWizardSettings.xml
-.idea/dictionaries
-.idea/libraries
-.idea/caches
+.idea/*
```
Comment (nelsonmestevao) [inline .gitignore]: I think this should be in you gitignore_global since it is not project specific.

File: src/components/Login.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    validateJson(json) {
+        if (json.data.hasOwnProperty('jwt')) {
+            deviceStorage.saveJWT("id_token", json.data.jwt);
+            this.props.newJWT(json.data.jwt);
+        }
+        else {
+            this.setState({error: json.data.error, loading: false});
+            console.log(json);
```
Comment (nelsonmestevao) [inline src/components/Login.js]: Delete this line.

File: src/components/Login.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    }
+
+    loginUser() {
+        const API_CALL = 'https://sei19-safira.herokuapp.com/api/auth/sign_in';
```
Comment (nelsonmestevao) [inline src/components/Login.js]: Move this to a `.env` file.

File: src/components/Login.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    render() {
+        const {email, password, error, loading} = this.state;
+        const {form, section, errorTextStyle} = styles;
+
+        return (
+            <Fragment>
+                <View style={form}>
+                    <View style={section}>
+                        <Input
+                            placeholder='patron@cesium.pt'
```
Comment (nelsonmestevao) [inline src/components/Login.js]: I don't really agree with this placeholder :smile:

File: src/components/Login.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        backgroundColor: '#fff',
+        borderColor: '#ddd',
+    },
+    errorTextStyle: {
+        alignSelf: 'center',
+        borderBottomWidth: 1,
+        fontSize: 18,
+        color: 'red'
+    }
+};
```
Comment (nelsonmestevao) [inline src/components/Login.js]: Remember the new line in the end.

File: src/screens/LoggedIn.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                </Text>
+                <Button
+                    onPress={this.props.deleteJWT}
+                    title="Log Out"
+                />
+            </View>
+
+        );
+    };
+}
```
Comment (nelsonmestevao) [inline src/screens/LoggedIn.js]: `\n` same

**Review by nelsonmestevao — CHANGES_REQUESTED** [review]
Check my comments and I think we can go. What do you have to say @apedrob ?

File: src/components/Login.js (hunk 4 above)
Comment (herulume) [reply src/components/Login.js]: 'youremail@domain' ?

File: src/components/Login.js (hunk 3 above)
Comment (herulume) [reply src/components/Login.js]: As far as I know to do this we another dependency. Is it really needed? 
We can point the API (safira) to a SEI's subdomain, not becoming specific to this SEI edition. 
@ZePedroResende @0urobor0s what do you think?

Comment (ZePedroResende) [reply src/components/Login.js]: Yes, i dont know if a .env is the standard in react native but a equivalent option is the best choice for easy deployment

**Comment by herulume** [comment]
Using .env file as of now. 
@ZePedroResende @nelsonmestevao @apedrob

**Review by nelsonmestevao — APPROVED** [review]
I would just suggest to move the IntelliJ files from the gitignore to the user gitignore_global.

The remaining :rocket:

**Pull request merged**